### PR TITLE
CU/7278-Chika-ReplaceWaygateNativeAlertFunctionWithCombo

### PR DESCRIPTION
**Pull request by cadibemma**
## Description of Change
- Update navigateTo() to reduce lines of code, simplify navigation and rope in waygate checks automatically.

## Screenshots/Video
- No UI Changes

## Testing

- [x] Tested on iOS 
- [x] Tested on Android 

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Comment by alexandec** [comment]
@cadibemma some conflicts to resolve on this one. Maybe it's related to the merge of the availability framework branch into develop?

**Comment by cadibemma** [comment]
> @cadibemma some conflicts to resolve on this one. Maybe it's related to the merge of the availability framework branch into develop?

yup! I'm working on it

File: VAMobile/src/screens/HomeScreen/ProfileScreen/PersonalInformationScreen/PersonalInformationScreen.tsx [hunk 1]
```diff
@@ -113,27 +114,19 @@ const PersonalInformationScreen: FC<PersonalInformationScreenProps> = ({ navigat
   const loadingCheck = loadingPersonalInfo || loadingGenderIdentityOptions || loadingDemographics
 
   const onGenderIdentity = () => {
```
Comment (alexandec) [inline VAMobile/src/screens/HomeScreen/ProfileScreen/PersonalInformationScreen/PersonalInformationScreen.tsx:116]: Probably no need for separate functions here anymore

File: VAMobile/src/screens/HomeScreen/ProfileScreen/ProfileScreen.tsx [hunk 2]
```diff
@@ -49,27 +49,19 @@ const ProfileScreen: FC<ProfileScreenProps> = ({ navigation }) => {
   }, [dispatch, militaryHistoryNeedsUpdate, userAuthorizedServices?.militaryServiceHistory, mhNotInDowntime])
 
   const onPersonalInformation = () => {
```
Comment (alexandec) [inline VAMobile/src/screens/HomeScreen/ProfileScreen/ProfileScreen.tsx:51]: Same thing with these separate functions

File: VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/SettingsScreen.tsx [hunk 3]
```diff
@@ -75,18 +74,16 @@ const SettingsScreen: FC<SettingsScreenProps> = ({ navigation }) => {
   }
 
   const onManage = () => {
```
Comment (alexandec) [inline VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/SettingsScreen.tsx:76]: Same for these separate functions

**Review by alexandec — CHANGES_REQUESTED** [review]
Nice work! Added a few minor suggestions, but fundamentally this looks good

File: VAMobile/src/screens/PaymentsScreen/PaymentsScreen.test.tsx [hunk 4]
```diff
@@ -90,23 +72,26 @@ context('PaymentsScreen', () => {
     it('should navigate to HowToUpdateDirectDeposit', () => {
       initializeTestInstance()
       fireEvent.press(screen.getByRole('menuitem', { name: 'Direct deposit information' }))
-      expect(navigateToHowToUpdateDirectDepositSpy).toHaveBeenCalled()
+      expect(mockNavigationSpy).toHaveBeenCalled()
```
[the remark this answers is not in the document]
Comment (cadibemma) [reply VAMobile/src/screens/PaymentsScreen/PaymentsScreen.test.tsx]: updated!

File: VAMobile/src/screens/HealthScreen/Pharmacy/PrescriptionHistory/PrescriptionHistory.tsx [hunk 5]
```diff
@@ -289,9 +288,7 @@ const PrescriptionHistory: FC<PrescriptionHistoryProps> = ({ navigation, route }
 
   const prescriptionDetailsClicked = (prescriptionID: string) => {
     logAnalyticsEvent(Events.vama_rx_details(prescriptionID))
-    if (waygateNativeAlert('WG_PrescriptionDetails')) {
-      return navigation.navigate('PrescriptionDetails', { prescriptionId: prescriptionID })
-    }
+    return navigation.navigate('PrescriptionDetails', { prescriptionId: prescriptionID })
```
[the remark this answers is not in the document]
Comment (cadibemma) [reply VAMobile/src/screens/HealthScreen/Pharmacy/PrescriptionHistory/PrescriptionHistory.tsx]: Updated everywhere except within CarouselTabBar. When I updated there it breaks the `'renders correctly through out each screen and calls completeFirstTimeLogin once you get to the end'` test in the OnboardingCarousel test file.

[the remark this answers is not in the document]
Comment (alexandec) [reply VAMobile/src/screens/HealthScreen/Pharmacy/PrescriptionHistory/PrescriptionHistory.tsx]: Ah I see, that test expects navigation to actually happen between the different carousel components. We have a global mock for useRouteNavigation in testSetup.ts which replaces it with a function that does nothing. You can override that in OnboardingCarousel.test.tsx with:

```
jest.mock('../../utils/hooks', () => ({
  ...jest.requireActual('../../utils/hooks'),
}))
```

That will place the original implementation back so navigation happens and the test succeeds.

[the remark this answers is not in the document]
Comment (cadibemma) [reply VAMobile/src/screens/HealthScreen/Pharmacy/PrescriptionHistory/PrescriptionHistory.tsx]: Thank you! It fixed the test

**Review by alexandec — DISMISSED** [review]

**Comment by alexandec** [comment]
Looks good!

**Comment by TKDickson** [comment]
Two questions for you @cadibemma:

- Claims and Generic Letter were intentionally _not_ enabled for availability framework, can you check with @dumathane about adding them? It's not clear to me whether, with the rest of the changes in here, we'll be able to intentionally exclude them still. If not, there's probably other stuff we need to add to this change.
- Can you update the testing scope in the ticket? Seems like it's "retest Use Case 1 for all existing screens" (which is what was in there originally) and also "do initial AF testing for the ~9 screens where it's now enabled" and.... idk, maybe more. You've got a lot of changes here :)

**Review by TKDickson — APPROVED** [review]
QA approves merging to develop

**Review by dumathane — APPROVED** [review]

**Pull request merged**